wikipedia/_tools/parse_clicks.py

```diff
@@ -0,0 +1,106 @@
+import csv
+import gzip
+import logging
+import os
+import pickle
+import sys
+from collections import Counter
+
+import requests
+
+# Set up the logger
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+# Define a handler to output log messages to the console
+console_handler = logging.StreamHandler()
+console_handler.setLevel(logging.INFO)
+
+# Define a formatter for the log messages
+formatter = logging.Formatter("%(asctime)s - %(levelname)s - %(message)s")
+console_handler.setFormatter(formatter)
+
+# Add the console handler to the logger
+logger.addHandler(console_handler)
+
+
+class ClickStreamDist:
+    def __init__(self, year=2023, month=6, lang="en"):
+        self.filename = f"clickstream-{lang}wiki-{year}-{month:02d}.tsv.gz"
+        self.url = f"https://dumps.wikimedia.org/other/clickstream/{year:d}-{month:02d}/{self.filename}"
+        self.clickstream_output_file = "clickstream.tsv.gz"
```

Comment:
Did you consider using `~/.rally/benchmarks/data/wikipedia/` to store the downloaded file so the source directory is not polluted with data file?

I would also prefer to keep the original file name so we can have several dataset downloaded.

Reply:
Updated